app/gui/src/dashboard/modals/TrialEndedForm.tsx

```diff
@@ -0,0 +1,60 @@
+/** @file Modal for handling user subscription after initial trial ended. */
+import { Alert } from '#/components/Alert'
+import { AlertDialog } from '#/components/AlertDialog'
+import { Text } from '#/components/Text'
+import * as backend from '#/services/Backend'
+import { useMutationCallback } from '#/utilities/tanstackQuery'
+import { useBackends } from '$/providers/backends'
+import { useText } from '$/providers/react'
+
+/** Props for a {@link TrialEndedModal}. */
+export interface TrialEndedModalProps {
+  readonly subscriptionId: backend.SubscriptionId
+}
+
+/** The mutation data for the `cancelSubscription` mutation. */
+interface CancelSubcriptionMutationParams {
+  readonly subscriptionId: backend.SubscriptionId
+}
+
+/** Modal for handling subscription after the trial period ended. */
+export function TrialEndedModal(props: TrialEndedModalProps) {
+  const { subscriptionId } = props
+  const { getText } = useText()
+  const { remoteBackend } = useBackends()
+
+  const onConfirm = useMutationCallback({
+    mutationFn: async () => {
+      const { url } = await remoteBackend.createCheckoutSession({
+        price: backend.Plan.solo,
+        quantity: 1,
+        interval: 1,
+        flowType: backend.BillingPortalFlowType.paymentMethodUpdate,
+      })
+      window.open(url, '_blank')?.focus()
+    },
+  })
+
+  const onCancel = useMutationCallback({
+    mutationFn: async (mutationData: CancelSubcriptionMutationParams) => {
```

Comment:
nit: this can be just a plain `backendMutationOptions`, would just need to do `onCancel([subscriptionId])` instead